### ISSUE TITLE
Introduction to Machine Learning - Proof Read (3_pipeline_task.ipynb)

### ISSUE THREAD

**Issue by sof202** [coding-for-reproducible-research/CfRR_Courses#238]
### One

https://github.com/coding-for-reproducible-research/CfRR_Courses/blob/70a1d0116a90794de11cd3ba586d1b3da47d569a/individual_modules/introduction_to_machine_learning/3_pipeline_task.ipynb#L50

`what we**'**re doing`

### Two

https://github.com/coding-for-reproducible-research/CfRR_Courses/blob/70a1d0116a90794de11cd3ba586d1b3da47d569a/individual_modules/introduction_to_machine_learning/3_pipeline_task.ipynb#L196

`never go **above** this value.`

### Three

https://github.com/coding-for-reproducible-research/CfRR_Courses/blob/70a1d0116a90794de11cd3ba586d1b3da47d569a/individual_modules/introduction_to_machine_learning/3_pipeline_task.ipynb#L366

`and **ensure** that there are`

### Four

https://github.com/coding-for-reproducible-research/CfRR_Courses/blob/70a1d0116a90794de11cd3ba586d1b3da47d569a/individual_modules/introduction_to_machine_learning/3_pipeline_task.ipynb#L377

`amount) **of** our training set` - Not 100% sure on this, but `from` sounds slightly wrong to me.

### Five

https://github.com/coding-for-reproducible-research/CfRR_Courses/blob/70a1d0116a90794de11cd3ba586d1b3da47d569a/individual_modules/introduction_to_machine_learning/3_pipeline_task.ipynb#L523

Slightly awkward wording, maybe the following is slightly better (I'm still not happy with the below either)

"These lines are a good example of the kinds of 'quirks' in data that could be learned by a machine learning model."

### Six

https://github.com/coding-for-reproducible-research/CfRR_Courses/blob/70a1d0116a90794de11cd3ba586d1b3da47d569a/individual_modules/introduction_to_machine_learning/3_pipeline_task.ipynb#L599

`If it has a complicated **series** of transformations`

### Seven(very minor)

https://github.com/coding-for-reproducible-research/CfRR_Courses/blob/70a1d0116a90794de11cd3ba586d1b3da47d569a/individual_modules/introduction_to_machine_learning/3_pipeline_task.ipynb#L740

Missing apostrophe in "can't"

### Eight

https://github.com/coding-for-reproducible-research/CfRR_Courses/blob/70a1d0116a90794de11cd3ba586d1b3da47d569a/individual_modules/introduction_to_machine_learning/3_pipeline_task.ipynb#L853

What is meant by 'multiplied by rows' here? A one-hot encoding just creates n more columns (not $n*\sharp rows$ more columns). What am I missing here?

### One

https://github.com/coding-for-reproducible-research/CfRR_Courses/blob/70a1d0116a90794de11cd3ba586d1b3da47d569a/individual_modules/introduction_to_machine_learning/3_pipeline_task.ipynb#L1015

Maybe add here that standardisation should be reserved for normally distributed data (or at least data that is roughly symmetrically located around the mean). A reason is listed for when you don't want to use min-max scaling is given with: "Crushes values into a small range if there is a major outlier". However, standardisation would also struggle in datasets with significantly skewed features (lots of outliers only on one side of the distribution). Some kind of caveat would be nice here at least (as no methodology is a catch all).

### Two

https://github.com/coding-for-reproducible-research/CfRR_Courses/blob/70a1d0116a90794de11cd3ba586d1b3da47d569a/individual_modules/introduction_to_machine_learning/3_pipeline_task.ipynb#L1333

In section 10, the linear model is applied to the test set and it is shown that the error is slightly worse than on the training set. A decision tree is also created in step 8 and a claim of overfitting is made (which is pretty obviously true). Would it be a good idea to show that this model is overfitted by applying it to the test set as well? Maybe this is already planned as an exercise.

### One

https://github.com/coding-for-reproducible-research/CfRR_Courses/blob/70a1d0116a90794de11cd3ba586d1b3da47d569a/individual_modules/introduction_to_machine_learning/3_pipeline_task.ipynb#L120-L124

This is a little awkwardly formatted no? Why not just combine the steps if they require so much back and forth? Or put both steps under the same heading? 

### Two

https://github.com/coding-for-reproducible-research/CfRR_Courses/blob/70a1d0116a90794de11cd3ba586d1b3da47d569a/individual_modules/introduction_to_machine_learning/3_pipeline_task.ipynb#L698-L714

I feel like the code here could be moved about a little to better convey intent. Maybe move both of the sense checking lines (`.head()` call and checking correct number of rows dropped) to the same code block? This might make it more obvious what is happening. The first block removes rows with null values, whilst the second block is in place to show that the operations did indeed work.

**Comment by FinleyGibson** [comment]
Fixed typos/langauge issues. But have left Re-ordering question for @simonkirbysimonkirby to decide.
